### PR TITLE
README: improve browser setup advice

### PR DESCRIPTION
**Pull request by dcousens**
As per https://github.com/bitcoinjs/bitcoinjs-lib/issues/314, this updates the README to include more up-to-date and idiomatic build advice for users who want to use the library from the browser.

Suggestions welcome and appreciated.

### CODE REVIEW

**Comment by dcousens** [comment]
@weilu I've updated the instructions.

I believe I've kept true to your intentions,

> the use of browserify directly without having its version managed by a dependency manager.
see https://github.com/bitcoinjs/bitcoinjs-lib/commit/df50393727979008098362b8ae9360686907cdcf

> buffer is a generic function that should not be living under the bitcoin namespace
see https://github.com/bitcoinjs/bitcoinjs-lib/commit/1d9bc2cc4c3468eb18de7dcc975963f7156cc7f0

I think the above makes it abundantly clear we are just helping the user understand how to user browserify, and not just providing a convenient hack.

File: README.md [hunk 1]
```diff
@@ -51,16 +51,32 @@ From the repo:
...
 
-From NPM:
+From your repository, create a `foobar.js` file
 
-    $ npm -g install bitcoinjs-lib browserify uglify-js
-    $ browserify -r bitcoinjs-lib -s bitcoin | uglifyjs > bitcoinjs.min.js
+``` javascript
+var foobar = {
+  base58: require('bs58'),
+  bitcoin: require('bitcoinjs-lib'),
```
Comment (weilu) [inline README.md]: it will complain that `bitcoinjs-lib` is not found as it has not yet been installed through npm

File: README.md [hunk 2]
```diff
@@ -51,16 +51,32 @@ From the repo:
...
+  ecurve: require('ecurve'),
+  BigInteger: require('bigi'),
+  Buffer: require('buffer')
+}
 
-After loading this file in your browser, you will be able to use the global `bitcoin` object.
+module.exports = bitcoin
```
Comment (weilu) [inline README.md]: `foobar` you mean? You also don't have to give it such an awkward name. 

```
module.exports = {
  base58: require('bs58'),
  bitcoin: require('bitcoinjs-lib'),
  ecurve: require('ecurve'),
  BigInteger: require('bigi'),
  Buffer: require('buffer')
}
```

File: README.md (hunk 1 above)
Comment (dcousens) [inline README.md]: Same with `bs58`, and any others.

Comment (weilu) [inline README.md]: So? the instruction is still incomplete

Comment (dcousens) [inline README.md]: Actually, it has been installed.

This is under the `## Setup` header, with `## Installation` above it.
Therefore, at this stage, it is assumed they have `bitcoinjs-lib` installed.

File: README.md [hunk 3]
```diff
@@ -44,23 +44,35 @@ If you are looking for the original, it is tagged as `0.1.3`. Unless you need it
...
 
-From the repository: Compile `bitcoinjs-min.js` with the following command:
+module.exports = foobar
+```
 
-    $ npm run-script compile
+Then, using browserify, compile `foobar.js` for use in the browser:
 
-From NPM:
+    $ browserify foobar.js -s foobar > dist/foobar.js
```
Comment (weilu) [inline README.md]:
```
mkdir node_sandbox
cd node_sandbox/
npm init
npm install bitcoinjs-lib
browserify foobar.js -s foobar > dist/foobar.js
-bash: dist/foobar.js: No such file or directory
```

Comment (weilu) [inline README.md]:
```
browserify foobar.js -s foobar > bundle.js
Error: Cannot find module 'bs58' from '/Users/luwei/workspace/node_sandbox'
    at /usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:50:17
    at process (/usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:119:43)
    at /usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:128:21
    at load (/usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:60:43)
    at /usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:66:22
    at /usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:21:47
    at Object.oncomplete (fs.js:107:15)
```

Comment (weilu) [inline README.md]: @dcousens Did I fail at reading the readme?

Comment (dcousens) [inline README.md]: @weilu try now after 642315e

**Comment by dcousens** [comment]
> Again, how do you control versions of those libraries?
That is a good point.  I guess it becomes obvious in your examples, but this problem is independent of this PR as it also affects the node set up advice.

How do we control users using `bigi@latest` in conjunction with our API instead of what we use internally?
The best we can do is either simplify our API such that common JS types are used or we ask the users to verify their dependencies against what we use internally.

**Comment by dcousens** [comment]
I'm going to merge this, but the point brought up by @weilu is quite valid.
I'll summarize and it make another issue.

``` javascript
How do we control users using bigi@latest in conjunction with our API instead of what we use internally?
The best we can do is either simplify our API such that common JS types are used for data interchange or we ask the users to verify their dependencies against what we use internally.
```

**Pull request merged**